quaddtype/quaddtype/src/umath.c

```diff
@@ -56,9 +56,18 @@ quad_multiply_resolve_descriptors(PyObject *self, PyArray_DTypeMeta *dtypes[],
     // The operand units can be used as-is; no casting required for quad types.
     Py_INCREF(given_descrs[0]);
     loop_descrs[0] = given_descrs[0];
+
+    if (given_descrs[1] == NULL) {
+        Py_INCREF(given_descrs[0]);
+        loop_descrs[1] = given_descrs[0];
+        return NPY_NO_CASTING;
+    }
+    Py_INCREF(given_descrs[1]);
+    loop_descrs[1] = given_descrs[1];
+
     Py_INCREF(given_descrs[1]);
     loop_descrs[1] = given_descrs[1];
-    return NPY_NO_CASTING;
+    return NPY_SAFE_CASTING;
```

Comment:
You know that you don't have any byte-swaps or so, so this is always NO_CASTING.  In principle, you could always return the singleton instance also.  (The only reason NumPy has more instances is byte-swapping and metadata, which you don't have to support.)

Bu it's fine to just delete it for this DType really.

The function currently doesn't fill in the output `loop_descrs[2]`?